mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodEventHandler.m

```diff
@@ -720,34 +757,33 @@ - (BOOL)deleteBackLegacy:(NSUInteger)n at:(NSUInteger) pos with:(NSRange) select
     if (self.contextBuffer != nil && (pos == 0 || pos == NSNotFound)) {
         pos = self.contextBuffer.length + n;
     }
-    
+
     if ([self.AppDelegate debugMode]) {
         NSLog(@"Using Legacy mode.");
         NSLog(@"pos = %lu", pos);
     }
-    
+
     if (pos >= n) {
         // n is now the number of delete-backs we need to post (plus one more if there is selected text)
         if ([self.AppDelegate debugMode]) {
             NSLog(@"Legacy mode: calling postDeleteBacks");
             if (_clientCanProvideSelectionInfo == No || _clientCanProvideSelectionInfo == Unreliable)
                 NSLog(@"Cannot trust client to report accurate selection length - assuming no selection.");
         }
-        
-        // Note: If pos is "not found", most likely the client can't accurately report the location. This might be
-        // dangerous, but for now let's go ahead and attempt to delete the characters we think should be there.
+
         if (_pendingBuffer != nil && [[self pendingBuffer] length] > 0) {
-            NSException* exception = [NSException
```

Comment:
Throwing an exception in an input method is messy, causes macOS to eventually (after 3 repeats?) stop loading the IM until a restart. Better that we log to console and pretend all is good ...